cl/api/tests.py

```diff
@@ -1187,40 +1224,119 @@ async def test_attorney_filters(self) -> None:
         docket = await Docket.objects.afirst()
         attorney = await Attorney.objects.afirst()
+        party = await sync_to_async(PartyFactory)(
+            docket=self.docket_2,
+            attorneys=[attorney],
+        )
         await sync_to_async(PartyTypeFactory.create)(
-            party=await sync_to_async(PartyFactory)(
-                docket=docket,
-                attorneys=[attorney],
-            ),
-            docket=docket,
+            party=party,
+            docket=self.docket_2,
         )
         self.q = {"docket__date_created__range": "2017-04-14,2017-04-15"}
         await self.assertCountInResults(1)
         self.q = {"docket__date_created__range": "2017-04-15,2017-04-16"}
         await self.assertCountInResults(0)
 
+        # Initial request: Fetch all related records
+        self.q = {"docket__id": self.docket.pk}
+        r = await self.async_client.get(self.path, self.q)
+        results = r.data["results"]
+        self.assertEqual(
+            len(results),
+            1,
+            msg=f"Expected 1, but got {len(results)}.\n\nr.data was: {r.data}",
+        )
+        # Verify record has expected number of parties
+        self.assertEqual(
+            len(results[0]["parties_represented"]),
+            2,
+            msg=f"Expected 2, but got {len(results[0]['parties_represented'])}.\n\nr.data was: {r.data}",
+        )
+
+        # Fetch attorney records for docket (repeat request with "filter_nested_results")
+        self.q = {"docket__id": self.docket.pk, "filter_nested_results": True}
+        r = await self.async_client.get(self.path, self.q)
+        results = r.data["results"]
+        self.assertEqual(
+            len(results),
+            1,
+            msg=f"Expected 1, but got {len(results)}.\n\nr.data was: {r.data}",
+        )
+        # Verify top-level record has single party (due to filter)
+        self.assertEqual(
+            len(results[0]["parties_represented"]),
+            1,
+            msg=f"Expected 1, but got {len(results[0]['parties_represented'])}.\n\nr.data was: {r.data}",
+        )
+        # Verify expected party is present in the parties_represented list
+        self.assertIn(
+            str(self.party.pk), results[0]["parties_represented"][0]["party"]
+        )
+
+        # Request using parties_represented lookup: Fetch all related records
+        self.q = {"parties_represented__docket__id": self.docket_2.pk}
+        r = await self.async_client.get(self.path, self.q)
+        results = r.data["results"]
+        self.assertEqual(
+            len(results),
+            1,
+            msg=f"Expected 1, but got {len(results)}.\n\nr.data was: {r.data}",
+        )
+        # Verify record has expected number of parties
+        self.assertEqual(
+            len(results[0]["parties_represented"]),
+            2,
+            msg=f"Expected 2, but got {len(results[0]['parties_represented'])}.\n\nr.data was: {r.data}",
+        )
+
```

Comment:
I observed a strange filtering behavior, which can be reproduced in this test by inserting the following lines here:

```suggestion
            await sync_to_async(RoleFactory.create)(
            party=party, docket=self.docket, attorney=self.attorney_2
        )
        self.q = {"docket__id": self.docket_2.pk}
        r = await self.async_client.get(self.path, self.q)
        results = r.data["results"]
        self.assertEqual(
            len(results),
            1,
            msg=f"Expected 1, but got {len(results)}.\n\nr.data was: {r.data}",
        )
```

It adds a new role involving `self.docket`, so when applying the filter `"docket__id": self.docket_2.pk` the new role doesn't modify the results from this filter and continues returning only one result.

However, the filter below returns different results:

```
 {
            "parties_represented__docket__id": self.docket_2.pk,
            "filter_nested_results": True,
        }
```

In this case, it matches the attorney linked to `self.docket_2.pk` and also matches `self.attorney_2`. However, this is odd because the `Role` created is linked to `self.docket`. This can be confirmed by checking `self.attorney_2` `parties_represented` key, which is empty. So, while the nested filter is working, the parent filter does not appear to be applied correctly. Is this behavior expected?

Reply:
@albertisfu I apologize for the misunderstanding. It's my fault. When you asked about the `docket` lookup in the `PartyFilter` class, I was under the impression that it was using the `role` table. However, it actually uses the `party_types` table for filtering.

the `parties_represented__docket__id` lookup should return attorneys whose represented parties are linked to docket_2 through the `partyType` model. The addition of the new role is causing the test to fail because the party record is linked to `docket_2`(check line 1277).

Reply:
> this is the expected behavior because, even though Attorney 30 doesn’t have a Role directly related to docket_id=8, it's shown because party=15 is related to docket_id=8 in a different Role. Due to the filter uses the PartyType table for filtering, correct?

This isn't quite right. You're getting those attorneys because Parties `15` and `14` are linked to Docket `8` through the `partyType` table. Once you use `parties_represented__docket__id`, you're no longer looking at the roles table. 

Can you please verify that Parties 14 and 15 are linked to Docket 8?

Reply:
Got it! Yeah Parties 14 and 15 are related to Docket 8 through `partyType` instances. 

Is that also why `filter_nested_results=true` is unable to filter `parties_represented` in attorney 30, because party 15 is linked to docket_id 8,  correct?

Reply:
> Is that also why filter_nested_results=true is unable to filter parties_represented in attorney 30, because party 15 is linked to docket_id 8, correct?

That's correct!